spec/win32/unit/certstore_spec.rb

```diff
@@ -108,6 +108,48 @@
       end
     end
 
+    context "When retrieve valid certificate" do
+      let (:store_name) { "my" }
+      let (:certificate_name) { 'GeoTrust Global CA' }
+      let (:retrieve) { { CERT_NAME_ATTR_TYPE: "GeoTrust Global CA", CERT_NAME_DNS_TYPE: "GeoTrust Global CA",
+        CERT_NAME_EMAIL_TYPE: "", CERT_NAME_FRIENDLY_DISPLAY_TYPE: "GeoTrust Global CA",
+        CERT_NAME_RDN_TYPE: "US, GeoTrust Inc., GeoTrust Global CA", CERT_NAME_SIMPLE_DISPLAY_TYPE: "GeoTrust Global CA",
+        CERT_NAME_UPN_TYPE: "", CERT_NAME_URL_TYPE: "" } }
+
+      before(:each) do
+        allow_any_instance_of(certbase).to receive(:cert_retrieve).and_return(retrieve)
+        allow_any_instance_of(certbase).to receive_message_chain(:CertFindCertificateInStore, :last).and_return(true)
+      end
+
+      it "return message of successful deletion" do
+        store = certstore.open(store_name)
+        retrive_cert = store.retrieve(certificate_name)
+        expect(retrive_cert).to eq(retrieve)
+      end
+    end
+
+    context "When retrieve invalid certificate" do
+      let (:store_name) { "my" }
+      let (:certificate_name) { "tmp_cert.mydomain.com" }
+      it "return message of `Cannot find certificate`" do
+        allow_any_instance_of(certbase).to receive(:CertFindCertificateInStore).and_return(false)
+        store = certstore.open(store_name)
+        delete_cert = store.retrieve(certificate_name)
+        expect(delete_cert).to eq("Cannot find certificate with name as `tmp_cert.mydomain.com`. Please re-verify certificate Issuer name")
+      end
+    end
+
+    context "When passing empty certificate_name to retrieve it" do
```

Comment:
`When passing empty certificate_name` is sufficient if we'll have a description block for `#cert_retrieve`.